src/jsonpath/json_path.rs

```diff
@@ -433,8 +434,24 @@ impl<'i, 'j, S: SelectValue> TermEvaluationResult<'i, 'j, S> {
         !self.eq(s)
     }
 
-    fn re(&self, _s: &Self) -> bool {
-        false
+    fn re_is_match(regex: &str, s: &str) -> bool {
+        Regex::new(regex).map_or_else(|_| false, |re| Regex::is_match(&re, s))
+    }
+
+    fn re_match(&self, s: &Self) -> bool {
+        match (self, s) {
+            (TermEvaluationResult::Value(v), TermEvaluationResult::Str(regex)) => {
```

Comment:
Can't the regex value be another part of the json? I mean can't it by itself be a Value?
Also can't the right side be a Str?
Also, what about float? int? did you check how they behave?

Reply:
Matching against a regexp which is stored in a JSON field seems a more advanced use case to me
And I left it aside for now. We can add it when needed, unless you find it mandatory.

Numbers currently do not match since we would need to convert them to strings, and, in addition, it can be done by ORing with numerical predicates (test `op_string_regexp_match` is filtering out numbers)

Reply:
I think it should be feature complete before merge so we should check those stuff.